pkg/frontend/v2/frontend.go

```diff
@@ -200,6 +218,22 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 		}
 	}
 
+	f.mtx.Lock()
+	if f.stopped {
+		f.mtx.Unlock()
+		return nil, fmt.Errorf("frontend not running")
+	}
+
+	f.inflightRequests++
+	f.mtx.Unlock()
```

Comment:
[super-nit] I would visually separate entrance and code done on leaving the method.

```suggestion
	f.inflightRequests++
	f.mtx.Unlock()

```